### PR TITLE
WIP Feature/cookie popup

### PR DESCRIPTION
**Pull request by teodormoq**

### CODE REVIEW

File: src/app/component/CookiePopup/CookiePopup.style.scss [hunk 1]
```diff
@@ -0,0 +1,45 @@
+:root {
+	--cookiepopup-bg: #{$white};
+}
+
+.CookiePopup {
+	display: flex;
+	width: 100vw;
```
Comment (Danpiel) [inline src/app/component/CookiePopup/CookiePopup.style.scss]: Codacy Issue found: [Expected indentation of 4 spaces (indentation)](https://app.codacy.com/app/ScandiPWA/base-theme/pullRequest?prid=4649838)

**Comment by Danpiel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 23
           

Complexity increasing per file
==============================
- src/app/component/CookiePopup/CookiePopup.component.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/ScandiPWA/base-theme/pullRequest?prid=4649838&bid=15570677)

File: src/app/component/CookiePopup/CookiePopup.style.scss [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	}
+
+	&-CTA {
+		margin-left: 0;
+
+		@include mobile {
+			margin-top: 20px;
+
+			button {
+				width: 100%;
```
Comment (Danpiel) [inline src/app/component/CookiePopup/CookiePopup.style.scss]: Codacy Issue found: [Expected indentation of 16 spaces (indentation)](https://app.codacy.com/app/ScandiPWA/base-theme/pullRequest?prid=4649838)

File: src/app/component/CookiePopup/CookiePopup.style.scss [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+	display: flex;
+	width: 100vw;
+	position: fixed;
+	left: 0;
+	bottom: 0;
+	align-items: center;
+	justify-content: space-between;
+	padding: 40px 60px;
+	background-color: var(--cookiepopup-bg);
+	z-index: 300;
```
Comment (Danpiel) [inline src/app/component/CookiePopup/CookiePopup.style.scss]: Codacy Issue found: [Expected indentation of 4 spaces (indentation)](https://app.codacy.com/app/ScandiPWA/base-theme/pullRequest?prid=4649838)

File: src/app/component/CookiePopup/CookiePopup.style.scss [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+    &-Content {
+        padding-right: 15px;
+    }
+
+	p {
+		max-width: 780px;
+	}
+
+	&-CTA {
```
Comment (Danpiel) [inline src/app/component/CookiePopup/CookiePopup.style.scss]: Codacy Issue found: [Expected indentation of 4 spaces (indentation)](https://app.codacy.com/app/ScandiPWA/base-theme/pullRequest?prid=4649838)

File: src/app/component/CookiePopup/CookiePopup.style.scss [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+		margin-left: 0;
+
+		@include mobile {
+			margin-top: 20px;
+
+			button {
```
Comment (Danpiel) [inline src/app/component/CookiePopup/CookiePopup.style.scss]: Codacy Issue found: [Expected indentation of 12 spaces (indentation)](https://app.codacy.com/app/ScandiPWA/base-theme/pullRequest?prid=4649838)

File: src/app/component/CookiePopup/CookiePopup.style.scss [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+	bottom: 0;
+	align-items: center;
+	justify-content: space-between;
+	padding: 40px 60px;
+	background-color: var(--cookiepopup-bg);
+	z-index: 300;
+
+    @include mobile {
+        display: block;
+		padding: 25px 40px;
```
Comment (Danpiel) [inline src/app/component/CookiePopup/CookiePopup.style.scss]: Codacy Issue found: [Expected indentation of 8 spaces (indentation)](https://app.codacy.com/app/ScandiPWA/base-theme/pullRequest?prid=4649838)

File: src/app/component/CookiePopup/CookiePopup.style.scss [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+	--cookiepopup-bg: #{$white};
+}
+
+.CookiePopup {
+	display: flex;
+	width: 100vw;
+	position: fixed;
+	left: 0;
+	bottom: 0;
+	align-items: center;
```
Comment (Danpiel) [inline src/app/component/CookiePopup/CookiePopup.style.scss]: Codacy Issue found: [Expected indentation of 4 spaces (indentation)](https://app.codacy.com/app/ScandiPWA/base-theme/pullRequest?prid=4649838)

File: src/app/component/CookiePopup/CookiePopup.style.scss [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+
+	&_isAccepted {
+		display: none;
+	}
+
+    &-Content {
+        padding-right: 15px;
+    }
+
+	p {
```
Comment (Danpiel) [inline src/app/component/CookiePopup/CookiePopup.style.scss]: Codacy Issue found: [Expected indentation of 4 spaces (indentation)](https://app.codacy.com/app/ScandiPWA/base-theme/pullRequest?prid=4649838)

File: src/app/component/CookiePopup/CookiePopup.style.scss [hunk 9]
```diff
@@ -0,0 +1,45 @@
+:root {
+	--cookiepopup-bg: #{$white};
+}
+
+.CookiePopup {
+	display: flex;
```
Comment (Danpiel) [inline src/app/component/CookiePopup/CookiePopup.style.scss]: Codacy Issue found: [Expected indentation of 4 spaces (indentation)](https://app.codacy.com/app/ScandiPWA/base-theme/pullRequest?prid=4649838)

File: src/app/component/CookiePopup/CookiePopup.style.scss [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+
+	&-CTA {
+		margin-left: 0;
+
+		@include mobile {
+			margin-top: 20px;
+
+			button {
+				width: 100%;
+			}
```
Comment (Danpiel) [inline src/app/component/CookiePopup/CookiePopup.style.scss]: Codacy Issue found: [Expected indentation of 12 spaces (indentation)](https://app.codacy.com/app/ScandiPWA/base-theme/pullRequest?prid=4649838)

File: src/app/component/CookiePopup/CookiePopup.style.scss [hunk 11]
```diff
@@ -0,0 +1,45 @@
+:root {
+	--cookiepopup-bg: #{$white};
```
Comment (Danpiel) [inline src/app/component/CookiePopup/CookiePopup.style.scss]: Codacy Issue found: [Expected indentation of 4 spaces (indentation)](https://app.codacy.com/app/ScandiPWA/base-theme/pullRequest?prid=4649838)

File: src/app/component/CookiePopup/CookiePopup.style.scss [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+    &-Content {
+        padding-right: 15px;
+    }
+
+	p {
+		max-width: 780px;
+	}
+
+	&-CTA {
+		margin-left: 0;
```
Comment (Danpiel) [inline src/app/component/CookiePopup/CookiePopup.style.scss]: Codacy Issue found: [Expected indentation of 8 spaces (indentation)](https://app.codacy.com/app/ScandiPWA/base-theme/pullRequest?prid=4649838)

File: src/app/component/CookiePopup/CookiePopup.style.scss [hunk 13]
```diff
@@ -0,0 +1,45 @@
...
+.CookiePopup {
+	display: flex;
+	width: 100vw;
+	position: fixed;
+	left: 0;
+	bottom: 0;
+	align-items: center;
+	justify-content: space-between;
+	padding: 40px 60px;
+	background-color: var(--cookiepopup-bg);
```
Comment (Danpiel) [inline src/app/component/CookiePopup/CookiePopup.style.scss]: Codacy Issue found: [Expected indentation of 4 spaces (indentation)](https://app.codacy.com/app/ScandiPWA/base-theme/pullRequest?prid=4649838)

File: src/app/component/CookiePopup/CookiePopup.style.scss [hunk 14]
```diff
@@ -0,0 +1,45 @@
+:root {
+	--cookiepopup-bg: #{$white};
+}
+
+.CookiePopup {
+	display: flex;
+	width: 100vw;
+	position: fixed;
```
Comment (Danpiel) [inline src/app/component/CookiePopup/CookiePopup.style.scss]: Codacy Issue found: [Expected indentation of 4 spaces (indentation)](https://app.codacy.com/app/ScandiPWA/base-theme/pullRequest?prid=4649838)

File: src/app/component/CookiePopup/CookiePopup.style.scss [hunk 15]
```diff
@@ -0,0 +1,45 @@
...
+	padding: 40px 60px;
+	background-color: var(--cookiepopup-bg);
+	z-index: 300;
+
+    @include mobile {
+        display: block;
+		padding: 25px 40px;
+    }
+
+	&_isAccepted {
```
Comment (Danpiel) [inline src/app/component/CookiePopup/CookiePopup.style.scss]: Codacy Issue found: [Expected indentation of 4 spaces (indentation)](https://app.codacy.com/app/ScandiPWA/base-theme/pullRequest?prid=4649838)

File: src/app/component/CookiePopup/CookiePopup.style.scss [hunk 16]
```diff
@@ -0,0 +1,45 @@
...
+
+.CookiePopup {
+	display: flex;
+	width: 100vw;
+	position: fixed;
+	left: 0;
+	bottom: 0;
+	align-items: center;
+	justify-content: space-between;
+	padding: 40px 60px;
```
Comment (Danpiel) [inline src/app/component/CookiePopup/CookiePopup.style.scss]: Codacy Issue found: [Expected indentation of 4 spaces (indentation)](https://app.codacy.com/app/ScandiPWA/base-theme/pullRequest?prid=4649838)

File: src/app/component/CookiePopup/CookiePopup.style.scss [hunk 17]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+.CookiePopup {
+	display: flex;
+	width: 100vw;
+	position: fixed;
+	left: 0;
+	bottom: 0;
+	align-items: center;
+	justify-content: space-between;
```
Comment (Danpiel) [inline src/app/component/CookiePopup/CookiePopup.style.scss]: Codacy Issue found: [Expected indentation of 4 spaces (indentation)](https://app.codacy.com/app/ScandiPWA/base-theme/pullRequest?prid=4649838)

File: src/app/component/CookiePopup/CookiePopup.style.scss [hunk 18]
```diff
@@ -0,0 +1,45 @@
...
+	&_isAccepted {
+		display: none;
+	}
+
+    &-Content {
+        padding-right: 15px;
+    }
+
+	p {
+		max-width: 780px;
```
Comment (Danpiel) [inline src/app/component/CookiePopup/CookiePopup.style.scss]: Codacy Issue found: [Expected indentation of 8 spaces (indentation)](https://app.codacy.com/app/ScandiPWA/base-theme/pullRequest?prid=4649838)

File: src/app/component/CookiePopup/CookiePopup.style.scss [hunk 19]
```diff
@@ -0,0 +1,45 @@
...
+	}
+
+	&-CTA {
+		margin-left: 0;
+
+		@include mobile {
```
Comment (Danpiel) [inline src/app/component/CookiePopup/CookiePopup.style.scss]: Codacy Issue found: [Expected indentation of 8 spaces (indentation)](https://app.codacy.com/app/ScandiPWA/base-theme/pullRequest?prid=4649838)

File: src/app/component/CookiePopup/CookiePopup.style.scss [hunk 20]
```diff
@@ -0,0 +1,45 @@
+:root {
+	--cookiepopup-bg: #{$white};
+}
+
+.CookiePopup {
+	display: flex;
+	width: 100vw;
+	position: fixed;
+	left: 0;
+	bottom: 0;
```
Comment (Danpiel) [inline src/app/component/CookiePopup/CookiePopup.style.scss]: Codacy Issue found: [Expected indentation of 4 spaces (indentation)](https://app.codacy.com/app/ScandiPWA/base-theme/pullRequest?prid=4649838)

File: src/app/component/CookiePopup/CookiePopup.style.scss [hunk 21]
```diff
@@ -0,0 +1,45 @@
...
+
+	p {
+		max-width: 780px;
+	}
+
+	&-CTA {
+		margin-left: 0;
+
+		@include mobile {
+			margin-top: 20px;
```
Comment (Danpiel) [inline src/app/component/CookiePopup/CookiePopup.style.scss]: Codacy Issue found: [Expected indentation of 12 spaces (indentation)](https://app.codacy.com/app/ScandiPWA/base-theme/pullRequest?prid=4649838)

File: src/app/component/CookiePopup/CookiePopup.style.scss [hunk 22]
```diff
@@ -0,0 +1,45 @@
+:root {
+	--cookiepopup-bg: #{$white};
+}
+
+.CookiePopup {
+	display: flex;
+	width: 100vw;
+	position: fixed;
+	left: 0;
```
Comment (Danpiel) [inline src/app/component/CookiePopup/CookiePopup.style.scss]: Codacy Issue found: [Expected indentation of 4 spaces (indentation)](https://app.codacy.com/app/ScandiPWA/base-theme/pullRequest?prid=4649838)

File: src/app/component/CookiePopup/CookiePopup.style.scss [hunk 23]
```diff
@@ -0,0 +1,45 @@
...
+	background-color: var(--cookiepopup-bg);
+	z-index: 300;
+
+    @include mobile {
+        display: block;
+		padding: 25px 40px;
+    }
+
+	&_isAccepted {
+		display: none;
```
Comment (Danpiel) [inline src/app/component/CookiePopup/CookiePopup.style.scss]: Codacy Issue found: [Expected indentation of 8 spaces (indentation)](https://app.codacy.com/app/ScandiPWA/base-theme/pullRequest?prid=4649838)

File: src/app/component/CookiePopup/CookiePopup.component.js [hunk 24]
```diff
@@ -0,0 +1,56 @@
...
+
+  	render() {
+		const { isAccepted } = this.state;
+
+		return (
+			<div block="CookiePopup" mods={ { isAccepted: isAccepted } }>
+				<div block="CookiePopup" elem="Content">
+					<p>
+						{ __('We use cookies, also from third parties, to remember your settings, statistics and targeted marketing. You accept all cookies by clicking further or by clicking "accept". Read more about our cookies and their purpose in our Cookie Policy - ') }
+						<Link to="/page/privacy-policy-cookie-restriction-mode">{ __('read more') }</Link>
```
Comment (alfredsgenkins) [inline src/app/component/CookiePopup/CookiePopup.component.js]: I think config defined page would be more preferable here.

File: src/app/component/CookiePopup/CookiePopup.component.js [hunk 25]
```diff
@@ -0,0 +1,56 @@
...
+		this.acceptCookies = this.acceptCookies.bind(this);
+	}
+
+	acceptCookies() {
+		BrowserDatabase.setItem(true, COOKIE_POPUP);
+		this.setState({ isAccepted: true });
+	}
+
+  	render() {
+		const { isAccepted } = this.state;
```
Comment (alfredsgenkins) [inline src/app/component/CookiePopup/CookiePopup.component.js]: Prefer rendering null if accepted.

File: src/app/component/CookiePopup/CookiePopup.component.js [hunk 26]
```diff
@@ -0,0 +1,56 @@
...
+
+class CookiePopup extends PureComponent {
+	constructor() {
+		super();
+
+		this.state = {
+			isAccepted: BrowserDatabase.getItem(COOKIE_POPUP) || false
+		};
+
+		this.acceptCookies = this.acceptCookies.bind(this);
```
Comment (alfredsgenkins) [inline src/app/component/CookiePopup/CookiePopup.component.js]: Prefer declaring this on class itself, no need for constructor for both state and acceptCookies function.

File: src/app/component/CookiePopup/CookiePopup.component.js [hunk 27]
```diff
@@ -0,0 +1,56 @@
...
+	}
+
+  	render() {
+		const { isAccepted } = this.state;
+
+		return (
+			<div block="CookiePopup" mods={ { isAccepted: isAccepted } }>
+				<div block="CookiePopup" elem="Content">
+					<p>
+						{ __('We use cookies, also from third parties, to remember your settings, statistics and targeted marketing. You accept all cookies by clicking further or by clicking "accept". Read more about our cookies and their purpose in our Cookie Policy - ') }
```
Comment (alfredsgenkins) [inline src/app/component/CookiePopup/CookiePopup.component.js]: I think this also can we taken from Magento config, but I am not sure.

**Comment by alfredsgenkins** [comment]
Thanks for your contribution! Please address questions from review :)

File: src/app/component/CookiePopup/CookiePopup.component.js (hunk 27 above)
Comment (teodormoq) [reply src/app/component/CookiePopup/CookiePopup.component.js]: The text is not possible as default to add from magento.
There is 2 thing we can do tho, either we make a rich text editor in the cookie settings section and make graphql support, or we can make it as a block.

It's up to you what you prefer?

Comment (alfredsgenkins) [reply src/app/component/CookiePopup/CookiePopup.component.js]: Let's add the `cookie notice` configuration filed (text) not supporting HTML in *Design / Configuration*. You can do this in [store-graphql](https://github.com/scandipwa/store-graphql). Expose it via GraphQL `storeConfig` type :) Thanks!

**Comment by Danpiel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/app/component/CookiePopup/CookiePopup.component.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/ScandiPWA/base-theme/pullRequest?prid=4649838&bid=15570677)

**Comment by Danpiel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/app/component/CookiePopup/CookiePopup.component.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ScandiPWA/base-theme/pullRequest?prid=4649838&bid=15570677)

**Pull request closed without merging**